test/built-ins/Promise/allSettled/resolved-sequence-mixed.js

```diff
@@ -37,22 +37,26 @@ sequence.push(1);
 
 p1.catch(function() {
   sequence.push(3);
+  assert.sameValue(sequence.length, 3);
   checkSequence(sequence, 'Expected to be called first.');
-}).catch($DONE);
+});
 
 Promise.allSettled([p1, p2, p3]).then(function() {
   sequence.push(6);
+  assert.sameValue(sequence.length, 6);
   checkSequence(sequence, 'Expected to be called fourth.');
 }).then($DONE, $DONE);
```

Comment:
It is interesting how this last promise catches all the errors in this test. sequence will make sure of the order. Those .catch($DONE) are really ok to be removed.